app/src/main/java/org/simple/clinic/activity/TheActivityLifecycle.kt

```diff
@@ -2,25 +2,27 @@ package org.simple.clinic.activity
 
 import org.simple.clinic.widgets.UiEvent
 
+private fun event(activityName: String?, event: String): String = activityName?.let { "$activityName:$event" } ?: ""
+
 sealed class TheActivityLifecycle : UiEvent {
 
-  class Resumed : TheActivityLifecycle() {
-    override val analyticsName = "TheActivity:Resumed"
+  class Resumed(activityName: String?) : TheActivityLifecycle() {
```

Comment:
Curious, what is the idea behind a nullable `activityName` parameter?

Reply:
Ah, it's nulllable if you don't want a particular activity's event to be reported to analytics.

Reply:
Please raise a pivotal ticket to address temporal coupling between these events and analytics.

Reply:
https://www.pivotaltracker.com/story/show/169281424